stats/members/mdetail.php

```diff
@@ -11,23 +11,32 @@
 
 require_login();
 
-$id = get_integer_param($_GET, 'id', null, 0, null);
-$user = User::load_from_uid($id);
+$id = get_integer_param($_GET, 'id', null, 0, null, true);
+if ($id === null) {
+    $user = User::load_current();
+} else {
+    $user = User::load_from_uid($id);
+}
 
 $valid_tally_names = array_keys(get_page_tally_names());
 $tally_name = get_enumerated_param($_GET, 'tally_name', null, $valid_tally_names, true);
 
 $can_reveal = can_reveal_details_about($user->username, $user->u_privacy);
-if ($can_reveal) {
-    $user_referent = "'" . $user->username . "'";
+if ($user == User::load_current()) {
+    $desc = _("Your user details");
 } else {
-    $user_referent = "#" . $user->u_id;
-    // Note that this doesn't reveal anything;
-    // the requestor already knows the subject's u_id,
-    // because it was included in the request.
+    if ($can_reveal) {
+        $user_referent = "'" . $user->username . "'";
```

Comment:
do we need html encoding? Apologies to keep bringing this up and causing issues with the sql statement for instance.

Reply:
The problems with the SQL statement were entirely mine because I didn't adequately test it and you were completely right about needing it, so please don't apologize. Given what happened some years ago I'm always interested in possible SQL escaping failures.

We should _probably_ be `html_safe()`ing some of the stuff in this file -- like `$desc` when outputting the heading -- but honestly I just don't have the gumption for changing unrelated code right now. I'm leaving it no worse off than I found it which is my minimum bar.

Related, I think we should revisit our philosophy on needing `html_safe()` around translated strings that don't contain user input. We manage the string translations like we do the code: only a very limited number of people have access to it and we can enforce correctness in them. Also, sometimes our translated strings contain HTML code (like links) that we can't escape which means we're being inconsistent with this right now anyway. We still need `attr_safe()` on all translated strings though as they can and will contain single quotes.

Reply:
Perfect. Works for me.